get-quiz-info.R

```diff
@@ -16,19 +16,45 @@ all_answers <- list()
 for (i in 1:length(data$assessment$section)) {
   ## For each question, get the question text:
   item <- data$assessment$section[[i]]
-  questions[i] <- item$presentation$material$mattext$text
-  answers <- list()
-  for (j in 1:length(item$presentation$response_lid$render_choice)) {
-    ## For all answers, get the answer text
-    response_label <- item$presentation$response_lid$render_choice[[j]]
-    answers[j] <- response_label$material$mattext$text
-    if (j == length(item$presentation$response_lid$render_choice)) {
-      ## Put the answers in a list
-      all_answers[[i]] <- unlist(answers)
+  if(class(item) != "character"){
+    questions[i] <- item$presentation$material$mattext$text
+    answers <- list()
+    for (j in 1:length(item$presentation$response_lid$render_choice)) {
+      ## For all answers, get the answer text
+      response_label <- item$presentation$response_lid$render_choice[[j]]
+      if(class(response_label$material$mattext) != "character"){
+        answers[j] <- response_label$material$mattext$text
+      }
+      if (j == length(item$presentation$response_lid$render_choice)) {
+        ## Put the answers in a list
+        all_answers[[i]] <- unlist(answers)
+      }
     }
   }
 }
 
+# in case not all questions have the same number of possible answers
+max_answers <- 0
+for(k in 1:length(all_answers)){
+  if(length(all_answers[[k]] > max_answers)){
+    max_answers <- length(all_answers[[k]])
+  }
+}
+for(l in 1:length(all_answers)){
+  if(length(all_answers[[l]] < max_answers)){
+    all_answers[[l]] <- c(all_answers[[l]], rep("", max_answers-length(all_answers[[l]])))
```

Comment:
I think if this is turned into a function, then yes, that should be an input option. I was thinking blank would be easier for whatever somebody did with the quiz file afterwards.